NextcloudTalk/RoomSearchTableViewController.m

```diff
@@ -70,43 +73,158 @@ - (void)didReceiveMemoryWarning
+        [_roomSearchBackgroundView.placeholderView setHidden:[self searchSections].count > 0];
+    }
+}
+
+- (void)showSearchingFooterView
+{
+    UIActivityIndicatorView *loadingMoreView = [[UIActivityIndicatorView alloc] initWithFrame:CGRectMake(0, 0, 44, 44)];
+    loadingMoreView.color = [UIColor darkGrayColor];
+    [loadingMoreView startAnimating];
+    self.tableView.tableFooterView = loadingMoreView;
+}
+
+- (void)hideSearchingFooterView
+{
+    self.tableView.tableFooterView = nil;
+}
+
+- (void)clearSearchedResults
+{
+    _rooms = @[];
+    _listableRooms = @[];
+    _messages = @[];
+    
+    [self reloadAndCheckSeachingIndicator];
 }
 
+
 #pragma mark - Utils
 
+- (NSArray *)searchSections
+{
+    NSMutableArray *sections = [NSMutableArray new];
+    if (_rooms.count > 0) {
+        [sections addObject:@(RoomSearchSectionFiltered)];
+    }
+    if (_listableRooms.count > 0) {
+        [sections addObject:@(RoomSearchSectionListable)];
+    }
+    if (_messages.count > 0) {
+        [sections addObject:@(RoomSearchSectionMessages)];
+    }
+    return [NSArray arrayWithArray:sections];
+}
+
 - (NCRoom *)roomForIndexPath:(NSIndexPath *)indexPath
 {
-    if (indexPath.section == RoomSearchSectionFiltered && indexPath.row < _rooms.count) {
+    NSInteger searchSection = [[[self searchSections] objectAtIndex:indexPath.section] integerValue];
+    if (searchSection == RoomSearchSectionFiltered && indexPath.row < _rooms.count) {
         return [_rooms objectAtIndex:indexPath.row];
-    } else if (indexPath.section == RoomSearchSectionListable && indexPath.row < _listableRooms.count) {
+    } else if (searchSection == RoomSearchSectionListable && indexPath.row < _listableRooms.count) {
         return [_listableRooms objectAtIndex:indexPath.row];
     }
     
     return nil;
 }
 
-- (BOOL)hasResults
+- (NCCSearchEntry *)messageForIndexPath:(NSIndexPath *)indexPath
 {
-    return _rooms.count > 0 || _listableRooms.count > 0;
+    NSInteger searchSection = [[[self searchSections] objectAtIndex:indexPath.section] integerValue];
+    if (searchSection == RoomSearchSectionMessages && indexPath.row < _messages.count) {
+        return [_messages objectAtIndex:indexPath.row];;
+    }
+    
+    return nil;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForMessageAtIndexPath:(NSIndexPath *)indexPath
+{
+    NCCSearchEntry *messageEntry = [_messages objectAtIndex:indexPath.row];
+    RoomTableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:kRoomCellIdentifier];
+    if (!cell) {
+        cell = [[RoomTableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:kRoomCellIdentifier];
+    }
+    
+    cell.titleLabel.text = messageEntry.title;
+    cell.subtitleLabel.text = messageEntry.subline;
+    [cell.roomImage setImageWithURLRequest:[[NSURLRequest alloc] initWithURL:[[NSURL alloc] initWithString:messageEntry.thumbnailURL]] placeholderImage:nil success:nil failure:nil];
```

Comment:
Use `(void)setImageWithURL:(NSURL *)url` to make this shorter?

Reply:
For this particular case I guess we can use `(void)setImageWithURL:(NSURL *)url` yes :)